server/hooks.go

```diff
@@ -85,6 +126,12 @@ func (p *Plugin) UserHasJoinedChannel(c *plugin.Context, channelMember *model.Ch
 		ChannelId: channelMember.ChannelId,
 		Message:   string(data),
 	}
-	time.Sleep(1 * time.Second)
+
+	if actor.Id == channelMember.UserId {
+		time.Sleep(3 * time.Second)
+	} else {
+		time.Sleep(15 * time.Second)
+	}
```

Comment:
What is the reasoning behind this change?

Reply:
This was added by another Dev I believe. However, I think it was due to a cluster related timing issue. I can look more into this and why it was added.

Reply:
So digging into this deeper, the reason this was added was to prevent a race condition between the posted ephemeral message as well as the actual welcome message. The reasoning from our Dev was: "there was some race condition involving the welcome message being sent in a specific case. I think in one instance of a user being added to a channel, the ephemeral message was sent before the user was added to the channel, so they could never see it."

However, I think this can easily be mitigated by removing the ephemeral message as this was the issue with redundancy found inside of the team welcome messages as well. What are your thoughts?